sunbeam-python/sunbeam/steps/openstack.py

```diff
@@ -152,6 +152,46 @@ def get_storage_tfvars(self, storage_nodes: list[dict]) -> dict:
 
         return tfvars
 
+    def get_mysql_storage_map_tfvars(self) -> dict:
+        """Return Mysql Storage map based on topology."""
+        if self.topology == "single":
+            # Leave the storage to defaults 10GB
+            return {}
+
+        # Do not return defaults if storage is set in manifest
+        mysql_k8s_from_manifest = self.manifest.core.software.charms.get("mysql-k8s")
+        if self.database == "single":
+            try:
+                if (
+                    mysql_k8s_from_manifest
+                    and mysql_k8s_from_manifest.model_extra
+                    and "storage" in mysql_k8s_from_manifest.model_extra
+                ):
+                    # Storage set in manifest, return
+                    return {}
+            except AttributeError:
+                pass
+
+            return {"mysql-storage": {"database": "20G"}}
+        else:
+            try:
+                if (
+                    mysql_k8s_from_manifest
+                    and mysql_k8s_from_manifest.model_extra
+                    and "storage-map" in mysql_k8s_from_manifest.model_extra
+                ):
+                    # Storage set in manifest, return
+                    return {}
+            except AttributeError:
+                pass
+
+            return {
+                "mysql-storage-map": {
+                    "nova": {"database": "10G"},
+                    "gnocchi": {"database": "10G"},
```

Comment:
I don't believe this is the correct way forward. We're trying not to reference features from outside the feature package if possible.

https://github.com/canonical/snap-openstack/pull/258 is an example on how to achieve that.

We should make sure the value does not change between manifest application, because this will return a failure from Juju and we should handle that gracefully.

Reply:
Fixed in subsequent push